pkg/operator/k8sutil/prometheus.go

```diff
@@ -86,3 +87,18 @@ func CreateOrUpdateServiceMonitor(ctx context.Context, serviceMonitorDefinition
 	}
 	return sm, nil
 }
+
+// DeleteServiceMonitor deletes a ServiceMonitor and returns the error if any
+func DeleteServiceMonitor(ctx context.Context, ns string, name string) error {
+	client, err := getMonitoringClient()
+	if err != nil {
+		return fmt.Errorf("failed to get monitoring client. %v", err)
+	}
+	e := client.MonitoringV1().ServiceMonitors(ns).Delete(ctx, name, metav1.DeleteOptions{})
```

Comment:
nit: for consistency name the error as `err`
```suggestion
	err = client.MonitoringV1().ServiceMonitors(ns).Delete(ctx, name, metav1.DeleteOptions{})
```